src/compo/viirs_l1bnc2ioda.py

```diff
@@ -170,7 +183,7 @@ def _read(self):
                 err = obsgrp.variables[errname]
                 err.set_auto_scale(False)
 
-                vals[:, ichan] = obs[:].data.ravel()
+                vals[:, ichan] = obs[:].data.ravel() * sec_term
```

Comment:
```suggestion
                vals[:, ichan] = obs[:].data.ravel() * sec_term if self.apply_secterm else obs[:].data.ravel()
```

Reply:
so break the conditionals into two sections one to compute secant exclusively

and one to apply it exclusively

there will likely be no performance hit as optimizers within the language would handle such a straightforward logic.  and it does make the readibility ever so slightly better